packages/document-converter/engine/document_converter/converter.py

```
@@ -0,0 +1,928 @@
+        temp_dir: Temporary directory where images should be copied.
+    """
+    # Find all image references in markdown
+    for img_match in re.finditer(r'!\[.*?\]\(([^)]+)\)', markdown_content):
+        img_path = img_match.group(1)
+
+        # Skip URLs
+        if img_path.startswith(('http://', 'https://', 'data:')):
+            continue
+
+        # Skip already processed mermaid diagrams (diagram_X.png or just filename.png in temp)
+        if img_path.startswith('diagram_') or not '/' in img_path:
```

Ruff (0.15.18)

[error] Test for membership should be `not in`

Convert to `not in`

(E713)